### PR TITLE
[MIG] Migrated 'base_setup' module into new api.

### PR DESCRIPTION
**Pull request by psa-odoo**
Task: https://www.odoo.com/web#id=12343&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.QNPxrphbRZJOP43X

### CODE REVIEW

File: addons/base_setup/models/base_setup.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        ('Member', 'Member'),
+        ('Patient', 'Patient'),
+        ('Partner', 'Partner'),
+        ('Donor', 'Donor'),
+        ('Guest', 'Guest'),
+        ('Tenant', 'Tenant')], string='How do you call a Customer',
+        required=True,
+        default='Customer')
+
+    @api.multi
```
Comment (vme-odoo) [inline addons/base_setup/models/base_setup.py]: remove mutli from here.

File: addons/base_setup/models/base_setup.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    def execute(self):
+        def _case_insensitive_replace(ref_string, src, value):
+            import re
+            pattern = re.compile(src, re.IGNORECASE)
+            return pattern.sub(_(value), _(ref_string))
+        IrModelFields = self.env['ir.model.fields']
+        IrUiMenu = self.env['ir.ui.menu']
+        IrActionsActWindow = self.env['ir.actions.act_window']
+        for o in self:
```
Comment (vme-odoo) [inline addons/base_setup/models/base_setup.py]: use proper variable name.

File: addons/base_setup/models/base_setup.py (hunk 1 above)
Comment (psa-odoo) [inline addons/base_setup/models/base_setup.py]: Done!

File: addons/base_setup/models/base_setup.py (hunk 2 above)
Comment (psa-odoo) [inline addons/base_setup/models/base_setup.py]: Done!

File: addons/base_setup/models/base_setup.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        if existing_translations:
+            existing_translations.write({'value': value})
+        else:
+            IrTranslation.create({'name': name, 'lang': UserLang, 'type': type, 'src': src, 'value': value, 'res_id': res_id})
+        return {}
+
+    @api.multi
+    def execute(self):
+        def _case_insensitive_replace(ref_string, src, value):
+            import re
```
Comment (hmo-odoo) [inline addons/base_setup/models/base_setup.py]: do not put import statement in function. It should be top of python file

File: addons/base_setup/models/base_setup.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        IrModelFields = self.env['ir.model.fields']
+        IrUiMenu = self.env['ir.ui.menu']
+        IrActionsActWindow = self.env['ir.actions.act_window']
+        # translate label of field
+        for field in IrModelFields.search([('field_description', 'ilike', 'Customer')]):
+            field_ref = field.model_id.model + ',' + field.name
+            self.make_translations(field_ref, 'field', field.field_description, _case_insensitive_replace(
+                field.field_description, 'Customer', self.partner))
+        # translate help tooltip of field
+        for obj in self.pool.models.values():
```
Comment (hmo-odoo) [inline addons/base_setup/models/base_setup.py]: use self.env instead of self.pool to get models

File: addons/base_setup/models/res_config.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    module_inter_company_rules = fields.Boolean(string='Manage Inter Company',
+        help="This installs the module inter_company_rules.\n Configure company rules to automatically create SO/PO when one of your company sells/buys to another of your company.")
+    company_share_partner = fields.Boolean(string='Share partners to all companies',
+        help="Share your partners to all companies defined in your instance.\n"
+             " * Checked : Partners are visible for every companies, even if a company is defined on the partner.\n"
+             " * Unchecked : Each company can see only its partner (partners where company is defined). Partners not related to a company are visible for all companies.")
+
+    @api.multi
+    def open_company(self):
+        return {
```
Comment (hmo-odoo) [inline addons/base_setup/models/res_config.py]: self.ensure_one()

File: addons/base_setup/models/res_config.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    def set_base_defaults(self):
+        if self.font:
+            User = self.env.user
+            font_name = self.font.name
+            User.company_id.write({
+                'font': self.font.id,
+                'rml_header': self._change_header(User.company_id.rml_header, font_name),
+                'rml_header2': self._change_header(User.company_id.rml_header2, font_name),
+                'rml_header3': self._change_header(User.company_id.rml_header3, font_name)})
+        return {}
```
Comment (hmo-odoo) [inline addons/base_setup/models/res_config.py]: really need to return empty dict ?

File: addons/base_setup/models/res_config.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+    @api.multi
```
Comment (hmo-odoo) [inline addons/base_setup/models/res_config.py]: really need decorator ?

File: addons/base_setup/models/res_config.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    def act_discover_fonts(self):
+        return self.env["res.font"].font_scan()
+
+    @api.multi
```
Comment (hmo-odoo) [inline addons/base_setup/models/res_config.py]: really need decorator ?

File: addons/base_setup/models/res_config.py [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+        return {
+            'company_share_partner': not bool(partner_rule.active)
+        }
+
+    @api.multi
+    def set_default_company_share_partner(self):
+        partner_rule = self.env.ref('base.res_partner_rule')
+        for wizard in self:
+            self.env['ir.rule'].browse(partner_rule.id).write(
+                {'active': not bool(wizard.company_share_partner)})
```
Comment (hmo-odoo) [inline addons/base_setup/models/res_config.py]: why need to casting into Boolean ?

File: addons/base_setup/models/base_setup.py [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    def execute(self):
+        def _case_insensitive_replace(ref_string, src, value):
+            import re
+            pattern = re.compile(src, re.IGNORECASE)
+            return pattern.sub(_(value), _(ref_string))
+        IrModelFields = self.env['ir.model.fields']
+        IrUiMenu = self.env['ir.ui.menu']
+        IrActionsActWindow = self.env['ir.actions.act_window']
+        # translate label of field
+        for field in IrModelFields.search([('field_description', 'ilike', 'Customer')]):
```
Comment (hmo-odoo) [inline addons/base_setup/models/base_setup.py]: why need to check terminology only for "Customer"? . what is purpose of partner selection field ?

**Comment by hmo-odoo** [comment]
make separate python file per model and also make separate view file per model

File: addons/base_setup/models/res_config.py (hunk 8 above)
Comment (psa-odoo) [inline addons/base_setup/models/res_config.py]: Yes, its needed.

File: addons/base_setup/models/base_config.py [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+        partner_rule = self.env.ref('base.res_partner_rule')
+        return {
+            'company_share_partner': not partner_rule.active
+        }
+
+    @api.multi
+    def set_default_company_share_partner(self):
+        self.ensure_one()
+        partner_rule = self.env.ref('base.res_partner_rule')
+        self.env['ir.rule'].browse(partner_rule.id).write(
```
Comment (hmo-odoo) [inline addons/base_setup/models/base_config.py]: do not browse again: partner_rule already record object.

File: addons/base_setup/models/base_config.py [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+        help="Share your partners to all companies defined in your instance.\n"
+             " * Checked : Partners are visible for every companies, even if a company is defined on the partner.\n"
+             " * Unchecked : Each company can see only its partner (partners where company is defined). Partners not related to a company are visible for all companies.")
+
+    @api.multi
+    def open_company(self):
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_window',
+            'name': 'Your Company',
```
Comment (hmo-odoo) [inline addons/base_setup/models/base_config.py]: 'Your Company' should be able to translate

File: addons/base_setup/base_setup.py [hunk 13]
```diff
@@ -1,79 +0,0 @@
...
-
-import json
-import cgi
-from openerp import tools
-from openerp.osv import fields, osv
-from openerp.tools.translate import _
-from lxml import etree
-
-# Specify Your Terminology will move to 'partner' module
-class specify_partner_terminology(osv.osv_memory):
```
Comment (hmo-odoo) [inline addons/base_setup/base_setup.py]: can you make separate commit to remove unused code ?

File: addons/base_setup/views/base_config_settings_view.xml [hunk 14]
```diff
@@ -1,27 +1,24 @@
...
-            <field name="name">General Settings</field>
+            <field name="name">base.config.settings.view.form</field>
             <field name="model">base.config.settings</field>
             <field name="arch" type="xml">
                 <form string="General Settings" class="oe_form_configuration">
                     <header>
-                        <button string="Apply" type="object" name="execute" class="oe_highlight"/>
-                        <button string="Cancel" type="object" name="cancel" class="oe_link"/>
+                        <button name="execute" string="Apply" type="object" class="oe_highlight"/>
+                        <button name="cancel" string="Cancel" type="object" class="oe_link"/>
```
Comment (WHPODOO) [inline addons/base_setup/views/base_config_settings_view.xml]: Is it really need difference ?
gifd

File: addons/base_setup/views/base_config_settings_view.xml [hunk 15]
```diff
@@ -1,27 +1,24 @@
...
         <record id="view_general_configuration" model="ir.ui.view">
-            <field name="name">General Settings</field>
+            <field name="name">base.config.settings.view.form</field>
             <field name="model">base.config.settings</field>
             <field name="arch" type="xml">
                 <form string="General Settings" class="oe_form_configuration">
                     <header>
-                        <button string="Apply" type="object" name="execute" class="oe_highlight"/>
-                        <button string="Cancel" type="object" name="cancel" class="oe_link"/>
+                        <button name="execute" string="Apply" type="object" class="oe_highlight"/>
```
Comment (WHPODOO) [inline addons/base_setup/views/base_config_settings_view.xml]: Is it really need difference ?
gifd

File: addons/base_setup/views/base_config_settings_view.xml [hunk 16]
```diff
@@ -1,27 +1,24 @@
...
+                        <button name="cancel" string="Cancel" type="object" class="oe_link"/>
                     </header>
                     <separator string="General Settings"/>
                     <div>
                         <label string="You will find more options in your company details: address for the header and footer, overdue payments texts, etc."/>
-                        <button type="object" name="open_company" string="Configure your company data" icon="fa-cogs" class="oe_inline oe_link"/>
+                        <button name="open_company" type="object" string="Configure your company data" icon="fa-cogs" class="oe_inline oe_link"/>
```
Comment (WHPODOO) [inline addons/base_setup/views/base_config_settings_view.xml]: Is it really need difference ?
gifd
